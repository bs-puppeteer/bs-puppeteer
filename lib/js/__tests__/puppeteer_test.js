// Generated by BUCKLESCRIPT VERSION 2.2.2, PLEASE EDIT WITH CARE
'use strict';

var Fs = require("fs");
var Jest = require("@glennsl/bs-jest/lib/js/src/jest.js");
var Path = require("path");
var Curry = require("bs-platform/lib/js/curry.js");
var Js_dict = require("bs-platform/lib/js/js_dict.js");
var Js_option = require("bs-platform/lib/js/js_option.js");
var Puppeteer = require("puppeteer");
var Js_primitive = require("bs-platform/lib/js/js_primitive.js");
var Page$BsPuppeteer = require("../src/Page.js");
var Launcher$BsPuppeteer = require("../src/Launcher.js");

var getElementValueJs = ( function (element) { return element.value; } );

var getLengthOfElementsJs = ( function (elements) { return elements.length; } );

var getElementOuterHTMLJs = ( function (element) { return element.outerHTML; } );

var getElementOuterHTMLJsPromise = ( function (element) { return Promise.resolve(element.outerHTML); } );

var fixturesPath = Path.resolve(Js_option.getWithDefault("", Js_primitive.undefined_to_opt(typeof (__dirname) === "undefined" ? undefined : (__dirname))), "../../../__tests__/fixtures/");

var testPagePath = Path.resolve(fixturesPath, "./testPage.html");

var testPageJsPath = Path.resolve(fixturesPath, "./testPage.js");

var testPageCssPath = Path.resolve(fixturesPath, "./testPage.css");

var testPageContent = Fs.readFileSync(testPagePath, "utf8");

var noSandbox = Launcher$BsPuppeteer.makeLaunchOptions(/* None */0, /* None */0, /* None */0, /* None */0, /* Some */[/* array */[
        "--no-sandbox",
        "--disable-setuid-sandbox"
      ]], /* None */0, /* None */0, /* None */0, /* None */0, /* None */0, /* None */0, /* None */0, /* None */0, /* None */0, /* () */0);

describe("Puppeteer", (function () {
        Jest.test("executablePath", (function () {
                return Jest.Expect[/* toContainString */11]("chromium", Jest.Expect[/* expect */0](Puppeteer.executablePath()));
              }));
        return Jest.test("defaultArgs()", (function () {
                      return Jest.Expect[/* toBeGreaterThan */5](0, Jest.Expect[/* expect */0](Puppeteer.defaultArgs().length));
                    }));
      }));

describe("Browser", (function () {
        var browser = [/* () */0];
        Jest.beforeAllPromise(/* None */0, (function () {
                return Puppeteer.launch(noSandbox).then((function (res) {
                              browser[0] = res;
                              return Promise.resolve(/* () */0);
                            }));
              }));
        Jest.test("wsEndpoint()", (function () {
                return Jest.Expect[/* toContainString */11]("ws://127.0.0.1:", Jest.Expect[/* expect */0](browser[0].wsEndpoint()));
              }));
        Jest.testPromise(/* None */0, "userAgent()", (function () {
                return browser[0].userAgent().then((function (userAgent) {
                              return Promise.resolve(Jest.Expect[/* toContainString */11]("HeadlessChrome", Jest.Expect[/* expect */0](userAgent)));
                            }));
              }));
        Jest.testPromise(/* None */0, "version()", (function () {
                return browser[0].version().then((function (version) {
                              return Promise.resolve(Jest.Expect[/* toMatchRe */15]((/^HeadlessChrome/), Jest.Expect[/* expect */0](version)));
                            }));
              }));
        return Jest.afterAllPromise(/* None */0, (function () {
                      return browser[0].close();
                    }));
      }));

describe("Page", (function () {
        var browser = [/* () */0];
        var page = [/* () */0];
        Jest.beforeAllPromise(/* None */0, (function () {
                return Puppeteer.launch(noSandbox).then((function (res) {
                                browser[0] = res;
                                return browser[0].newPage();
                              })).then((function (res) {
                              page[0] = res;
                              return Promise.resolve(/* () */0);
                            }));
              }));
        Jest.testPromise(/* None */0, "$()", (function () {
                return page[0].$("body").then((function (elementHandle) {
                              return Promise.resolve(Jest.ExpectJs[/* toBeTruthy */28](Jest.Expect[/* expect */0](elementHandle)));
                            }));
              }));
        Jest.testPromise(/* None */0, "content()", (function () {
                return page[0].content().then((function (content) {
                              return Promise.resolve(Jest.ExpectJs[/* toBe */2]("<html><head></head><body></body></html>", Jest.Expect[/* expect */0](content)));
                            }));
              }));
        Jest.testPromise(/* None */0, "$$()", (function () {
                return page[0].$$("html,body").then((function (elementHandles) {
                              return Promise.resolve(Jest.ExpectJs[/* toHaveLength */13](2, Jest.Expect[/* expect */0](elementHandles)));
                            }));
              }));
        Jest.testPromise(/* None */0, "$x", (function () {
                return page[0].$x("/html/body").then((function (elementHandles) {
                              return Promise.resolve(Jest.ExpectJs[/* toHaveLength */13](1, Jest.Expect[/* expect */0](elementHandles)));
                            }));
              }));
        Jest.testPromise(/* None */0, "$$eval()", (function () {
                return page[0].$$eval("html,body", Curry.__1(getLengthOfElementsJs)).then((function (length) {
                              return Promise.resolve(Jest.Expect[/* toBe */2](2.0, Jest.Expect[/* expect */0](length)));
                            }));
              }));
        Jest.testPromise(/* None */0, "$eval() with 0 args", (function () {
                return page[0].$eval("html", Curry.__1(getElementOuterHTMLJs)).then((function (html) {
                              return Promise.resolve(Jest.Expect[/* toBe */2]("<html><head></head><body></body></html>", Jest.Expect[/* expect */0](html)));
                            }));
              }));
        Jest.testPromise(/* None */0, "$eval() with 0 args returning a promise", (function () {
                return page[0].$eval("html", Curry.__1(getElementOuterHTMLJsPromise)).then((function (h) {
                              return Promise.resolve(Jest.Expect[/* toBe */2]("<html><head></head><body></body></html>", Jest.Expect[/* expect */0](h)));
                            }));
              }));
        Jest.testPromise(/* None */0, "$eval() with 1 arg", (function () {
                return page[0].setContent(testPageContent).then((function () {
                                var partial_arg = ( function (el, prop) { return el.getAttribute(prop); } );
                                return page[0].$eval("input", Curry.__2(partial_arg), "id");
                              })).then((function (id) {
                              return Promise.resolve(Jest.Expect[/* toBe */2]("input", Jest.Expect[/* expect */0](id)));
                            }));
              }));
        Jest.testPromise(/* None */0, "click()", (function () {
                return page[0].click("body", undefined).then((function () {
                              return Promise.resolve(Jest.pass);
                            }));
              }));
        Jest.testPromise(/* None */0, "goto()", (function () {
                return browser[0].newPage().then((function (page) {
                                  var options = {
                                    timeout: 25000
                                  };
                                  return page.goto("https://google.com", options);
                                })).then((function (res) {
                                return res.text();
                              })).then((function (text) {
                              return Promise.resolve(Jest.Expect[/* toContainString */11]("<title>Google</title>", Jest.Expect[/* expect */0](text)));
                            }));
              }));
        Jest.testPromise(/* None */0, "screenshot()", (function () {
                return page[0].screenshot(undefined).then((function (buf) {
                              return Promise.resolve(Jest.Expect[/* toBeGreaterThanOrEqual */6](3236, Jest.Expect[/* expect */0](buf.byteLength)));
                            }));
              }));
        Jest.testPromise(/* None */0, "waitForSelector()", (function () {
                return page[0].waitForSelector("body", undefined).then((function () {
                              return Promise.resolve(Jest.pass);
                            }));
              }));
        Jest.testPromise(/* None */0, "waitForXPath()", (function () {
                return page[0].waitForXPath("/html/body", Page$BsPuppeteer.makeSelectorOptions(/* None */0, /* None */0, /* Some */[100], /* () */0)).then((function (elementHandle) {
                              return Promise.resolve(Jest.ExpectJs[/* toBeTruthy */28](Jest.Expect[/* expect */0](elementHandle)));
                            }));
              }));
        Jest.testPromise(/* None */0, "setExtraHTTPHeaders", (function () {
                return page[0].setExtraHTTPHeaders(Js_dict.fromList(/* :: */[
                                  /* tuple */[
                                    "extra-http-header",
                                    "header01"
                                  ],
                                  /* [] */0
                                ])).then((function () {
                              return Promise.resolve(Jest.pass);
                            }));
              }));
        Jest.testPromise(/* None */0, "type()", (function () {
                return browser[0].newPage().then((function (page) {
                                return page.setContent(testPageContent).then((function () {
                                                return page.type("#input", "hello world", undefined);
                                              })).then((function () {
                                              return page.$eval("#input", Curry.__1(getElementValueJs));
                                            }));
                              })).then((function (value) {
                              return Promise.resolve(Jest.Expect[/* toBe */2]("hello world", Jest.Expect[/* expect */0](value)));
                            }));
              }));
        Jest.testPromise(/* None */0, "addScriptTag()", (function () {
                return browser[0].newPage().then((function (page) {
                              return page.addScriptTag({
                                              path: testPageJsPath
                                            }).then((function () {
                                              return page.content();
                                            })).then((function (content) {
                                            return page.close().then((function () {
                                                          return Promise.resolve(Jest.Expect[/* toContainString */11]("// This is \"testPage.js\"", Jest.Expect[/* expect */0](content)));
                                                        }));
                                          }));
                            }));
              }));
        Jest.testPromise(/* None */0, "addStyleTag()", (function () {
                return browser[0].newPage().then((function (page) {
                              return page.addStyleTag({
                                              path: testPageCssPath
                                            }).then((function () {
                                              return page.content();
                                            })).then((function (content) {
                                            return page.close().then((function () {
                                                          return Promise.resolve(Jest.Expect[/* toContainString */11]("/* This is \"testPage.css\" */", Jest.Expect[/* expect */0](content)));
                                                        }));
                                          }));
                            }));
              }));
        Jest.testPromise(/* None */0, "authenticate()", (function () {
                return page[0].authenticate({
                              username: "foo",
                              password: "bar"
                            }).then((function () {
                              return Promise.resolve(Jest.pass);
                            }));
              }));
        Jest.testPromise(/* None */0, "cookies()", (function () {
                return page[0].setCookie({
                                name: "foo",
                                value: "bar",
                                url: "http://localhost"
                              }).then((function () {
                                return page[0].cookies("http://localhost");
                              })).then((function (cookies) {
                              return Promise.resolve(Jest.Expect[/* toHaveLength */13](1, Jest.Expect[/* expect */0](cookies)));
                            }));
              }));
        Jest.testPromise(/* None */0, "setCookie()", (function () {
                return page[0].setCookie({
                                name: "foo",
                                value: "bar",
                                url: "http://localhost"
                              }, {
                                name: "foo2",
                                value: "bar2",
                                url: "http://localhost2"
                              }).then((function () {
                                return page[0].cookies("http://localhost", "http://localhost2");
                              })).then((function (cookies) {
                              return Promise.resolve(Jest.Expect[/* toHaveLength */13](2, Jest.Expect[/* expect */0](cookies)));
                            }));
              }));
        Jest.testPromise(/* None */0, "deleteCookie()", (function () {
                return page[0].setCookie({
                                  name: "foo",
                                  value: "bar",
                                  url: "http://localhost"
                                }).then((function () {
                                  return page[0].deleteCookie();
                                })).then((function () {
                                return page[0].cookies();
                              })).then((function (cookies) {
                              return Promise.resolve(Jest.Expect[/* toHaveLength */13](0, Jest.Expect[/* expect */0](cookies)));
                            }));
              }));
        Jest.testPromise(/* None */0, "emulate()", (function () {
                return page[0].emulate({
                              viewport: {
                                width: 320,
                                height: 480,
                                deviceScaleFactor: 2,
                                isMobile: true,
                                hasTouch: true,
                                isLandscape: true
                              },
                              userAgent: ""
                            }).then((function () {
                              return Promise.resolve(Jest.ExpectJs[/* toMatchObject */31]({
                                              width: 320,
                                              height: 480,
                                              deviceScaleFactor: 2,
                                              isMobile: true,
                                              hasTouch: true,
                                              isLandscape: true
                                            }, Jest.Expect[/* expect */0](page[0].viewport())));
                            }));
              }));
        Jest.testPromise(/* None */0, "emulateMedia()", (function () {
                return page[0].emulateMedia("print").then((function () {
                              return Promise.resolve(Jest.pass);
                            }));
              }));
        Jest.testPromise(/* None */0, "emulateMediaDisable()", (function () {
                return page[0].emulateMedia((null)).then((function () {
                              return Promise.resolve(Jest.pass);
                            }));
              }));
        Jest.testPromise(/* None */0, "evaluate()", (function () {
                var $$eval = function () {
                  return "ok";
                };
                return page[0].evaluate($$eval).then((function (res) {
                              return Promise.resolve(Jest.Expect[/* toBe */2]("ok", Jest.Expect[/* expect */0](res)));
                            }));
              }));
        Jest.testPromise(/* None */0, "evaluate() with 1 arg", (function () {
                var $$eval = function (arg) {
                  return arg + "iedoke";
                };
                return page[0].evaluate($$eval, "ok").then((function (res) {
                              return Promise.resolve(Jest.Expect[/* toBe */2]("okiedoke", Jest.Expect[/* expect */0](res)));
                            }));
              }));
        Jest.testPromise(/* None */0, "evaluate1() and a curried function", (function () {
                return page[0].evaluate((function (param) {
                                var arg1 = "hello";
                                var arg2 = param;
                                return arg1 + (" " + arg2);
                              }), "world").then((function (res) {
                              return Promise.resolve(Jest.Expect[/* toBe */2]("hello world", Jest.Expect[/* expect */0](res)));
                            }));
              }));
        Jest.testPromise(/* None */0, "evaluatePromise1() with a curried function", (function () {
                return page[0].evaluate((function (param) {
                                var arg1 = "hello";
                                var arg2 = param;
                                return Promise.resolve(arg1 + (" " + arg2));
                              }), "world").then((function (res) {
                              return Promise.resolve(Jest.Expect[/* toBe */2]("hello world", Jest.Expect[/* expect */0](res)));
                            }));
              }));
        Jest.testPromise(/* None */0, "evaluatePromise2() with a curried function", (function () {
                return page[0].evaluate((function (param, param$1) {
                                var arg1 = "hello";
                                var arg2 = param;
                                var arg3 = param$1;
                                return Promise.resolve(arg1 + (" " + (arg2 + arg3)));
                              }), "world", "!").then((function (res) {
                              return Promise.resolve(Jest.Expect[/* toBe */2]("hello world!", Jest.Expect[/* expect */0](res)));
                            }));
              }));
        Jest.testPromise(/* None */0, "evaluate() with 2 args", (function () {
                var $$eval = function (arg1, arg2) {
                  return arg1.length.toString() + (arg1 + (" " + arg2));
                };
                return page[0].evaluate($$eval, "hello", "world").then((function (res) {
                              return Promise.resolve(Jest.Expect[/* toBe */2]("5hello world", Jest.Expect[/* expect */0](res)));
                            }));
              }));
        Jest.testPromise(/* None */0, "evaluateString()", (function () {
                return page[0].setContent(testPageContent).then((function () {
                              return page[0].evaluate(" document.getElementsByTagName(\"title\")[0].innerHTML; ").then((function (title) {
                                            return Promise.resolve(Jest.Expect[/* toBe */2]("Test Page", Jest.Expect[/* expect */0](title)));
                                          }));
                            }));
              }));
        Jest.testPromise(/* None */0, "evaluateHandle()", (function () {
                var $$eval = function () {
                  return ( document );
                };
                return page[0].evaluateHandle($$eval).then((function (jsHandler) {
                              return Promise.resolve(Jest.ExpectJs[/* toBeTruthy */28](Jest.Expect[/* expect */0](jsHandler)));
                            }));
              }));
        return Jest.afterAllPromise(/* None */0, (function () {
                      return page[0].close().then((function () {
                                    return browser[0].close();
                                  }));
                    }));
      }));

exports.getElementValueJs = getElementValueJs;
exports.getLengthOfElementsJs = getLengthOfElementsJs;
exports.getElementOuterHTMLJs = getElementOuterHTMLJs;
exports.getElementOuterHTMLJsPromise = getElementOuterHTMLJsPromise;
exports.fixturesPath = fixturesPath;
exports.testPagePath = testPagePath;
exports.testPageJsPath = testPageJsPath;
exports.testPageCssPath = testPageCssPath;
exports.testPageContent = testPageContent;
exports.noSandbox = noSandbox;
/* getElementValueJs Not a pure module */
