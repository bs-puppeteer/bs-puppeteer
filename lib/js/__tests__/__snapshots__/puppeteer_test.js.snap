// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Coverage startCSSCoverage, stopCSSCoverage 1`] = `
Array [
  Object {
    "ranges": Array [
      Object {
        "end": 26,
        "start": 3,
      },
    ],
    "text": "
		input { color: green; }
		a { color: blue; }
		",
    "url": "file:///home/jihchi/repos/bs-puppeteer/__tests__/fixtures/testPage.html",
  },
]
`;

exports[`Coverage startJSCoverage, stopJSCoverage 1`] = `
Array [
  Object {
    "ranges": Array [
      Object {
        "end": 19,
        "start": 0,
      },
      Object {
        "end": 63,
        "start": 37,
      },
    ],
    "text": "
		function foo() {function bar() { } console.log(1); } foo(); ",
    "url": "file:///home/jihchi/repos/bs-puppeteer/__tests__/fixtures/testPage.html",
  },
]
`;
